### ISSUE TITLE
InvalidOperationInCellEditMode: Excel is in cell-editing mode. Please exit the edit mode by pressing ENTER or TAB or selecting another cell, and then try again.

### ISSUE THREAD

**Issue by YasharHeydari** [OfficeDev/office-js#3388]
# Provide required information needed to triage your issue

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: PC desktop
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: Microsoft® Excel® for Microsoft 365 MSO (16.0.14326.20782) 64-bit 
* Operating System: Windows 10
* Browser (if using Office on the web): ______
 
## Expected behavior

When we enter the edit mode on a cell, clicking on task pane should make the cell get out of edit-mode, because edit-mode blocks all other behaviors in Excel environment.

## Current behavior

When I click on a cell to enter edit mode and then I click on the task pane, on which we have a button that will run commands to set the cell value, I still get the same "InvalidOperationInCellEditMode error: Excel is in cell-editing mode" error! Shouldn't clicking on task pane make the excel exit from cell-editing mode?

## Steps to reproduce

1. Having a Office add-in task pane, create a button and put this code behind the button ( The code is changed for simplification)
   await Excel.run(async (context) => {
      const sheet = context.workbook.worksheets.getActiveWorksheet();
      const cell = sheet.getCell(1, 1);
      cell.values = [["something"]];

      await context.sync();
    });
3. Select any cell to enter edit-mode
4. Then Click on the button in the task pane
5. You'll get this error! 

## Link to live example(s)

1. ______
2. ______
3. ______

# Provide additional details

1. ______
2. ______
3. ______

## Context

Our clients cannot detect that it's in cell-edit mode, so they expect the add-in work fine, but it's doing nothing because of the edit-mode. 

## Useful logs

- [ ] Console errors
- [ ] Screenshots
- [ ] Test file (if only happens on a particular file)

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by honxmsft** [comment]
It seems this issue does not repro in new version of Excel. If you have to use the `16.0.14326.20782`, we can take a look in detail.

**Comment by YasharHeydari** [comment]
Yes please! This version is  used vastly by us and our clients, so could you please take a look into this version and if possible specify a version that this is explicitly addressed in ?

**Comment by YasharHeydari** [comment]
Any updates?

**Comment by honxmsft** [comment]
The bug does repro in old version, and in the new version, this issue is fixed.

If you just cannot update the version, I will send a request to backport the fix to older version to the related feature crew. Please notice the backporting has some criteria, if the issue does not satisfy the criteria, the team won't fix this in old version.

**Comment by YasharHeydari** [comment]
Could you please create a backport request?

**Comment by SiruiSun-MSFT** [comment]
Hi @YasharHeydari, Regarding the issue you mentioned, I would like to ask for some details. 

-Since this bug has been fixed in the new version, is there any blocking issue to prevent you from upgrading to the new version? 
-And could you provide your user scale to us? 

This will help us measure whether the work meets the backport criteria. Thanks!

**Comment by manuelsidler** [comment]
@SiruiSun-MSFT  I have the same issue (tested with Excel on Mac Version 16.81). We're not able to execute any commands when a user opens the task pane while editing a cell. It works if the task pane is already open and the user clicks something on the task pane.

**Comment by ErwinSalas** [comment]
This same issue is happening also in excel 2019
In my case I basically need to finish the cell edit mode when the user resume the activity in the addin, since I need to get the change that the user made and update the state of my addin.

```
selectCellBelowActiveCell = async (): Promise<void> => {
    try {
      await Excel.run(async (context) => {
        const sheet = context.workbook.worksheets.getActiveWorksheet()
        const range = sheet.getRange('A1') // Get a range to ensure context is active
        range.select()
        await context.sync()
      })
    } catch (error) {
      console.log(error.toString()) here is when it is triggering the error "InvalidOperationInCellEditMode: Excel is in cell-editing mode. Please exit the edit mode by pressing ENTER or TAB or selecting another cell, and then try again."
    }
  }
```
Does anyone have a workaround for this?

**Comment by duane-j-wagner** [comment]
I'm having similar trouble.  I stumbled on an old VB solution at the end of [this](https://www.vbforums.com/showthread.php?317660-Excel-Exit-Edit-Mode-RESOLVED-finally-P). I wish Excel would remove this restriction.  It seems silly that an add-in is blocked from editing cells just because a user put a cell (even in a different window) into edit mode.  Scripts should be treated as an independent user that can make changes at the same time as real users.
Calling Excel.run with the option set like so will at least prevent the exceptions:
`await Excel.run({delayForCellEdit: true}, async (context: Excel.RequestContext) => {...}`

**Comment by chandpravandan** [comment]
closing this issue since the reported scenario was fixed in the latest version of Excel on Desktop

you can request for a new feature by submitting the details here - https://techcommunity.microsoft.com/t5/microsoft-365-developer-platform/idb-p/Microsoft365DeveloperPlatform
